yktracec/src/jitmodbuilder.cc

```diff
@@ -828,6 +842,10 @@ class JITModBuilder {
     assert(CurFrame);
     CurFrame->setResume(CurBBIdx, Instr, CurInstrIdx);
 
+    // Clone the CallStack, store it on the heap, and hardcode its pointer
```

Comment:
Why do we clone the whole callstack?

Reply:
Will update the comment: we need it to initialise the jitmodbuilder when tracing after a guardfailure. Otherwise, when a guard fails inside a guard, we don't know where to continue tracing when that call returns.

Reply:
Fixed in a7f07f7a358

Reply:
Maybe I'm not understanding what `CallStack` is, but why do we need to duplicate it? Is it the C call stack or the shadow stack or some frames or ... ?

Reply:
It's the JITModBuilder call stack. Remember that we keep saying that JITModBuilder is not unlike an interpreter. Whenever we see a call in the trace, we need to push some info onto the `CallStack` that keeps track of where this call happened (and some other info) so that we can continue copying instructions at the correct block and instruction after we've finished inlining the call.

Now when we collect a side-trace, we are starting in the middle of a function. So when that function returns we have no idea where to continue collecting instructions from. PT only tells us the block but not the instruction. We do collect this information when collecting the parent trace, so we just need to clone it and then pass it to the side-trace.

Again, this is all a consequence of calls not being terminators in LLVM IR.

Reply:
Ah, I get it!